File: dotnet/ArqLibrarianClassic/ArcLibrarianClassis.ATests/Searching/SearchAll.cs
using ArcLibrarianClassis.ATests.Library;
using NUnit.Framework;

namespace ArcLibrarianClassis.ATests.Searching
{
    [TestFixture]
    public class SearchAll
    {
        private LibraryFixture fixture = null;
        
        [SetUp]
        public void SetupFixture()
        {
            fixture = new LibraryFixture();
        }  
        
        [Test]
        public void ShouldSearchAllBooks()
        {
            fixture.ApplicationStarted();
            fixture.HasSampleBooks();

            fixture.UserEnters("search");
            
            fixture.Then();
            fixture.SystemShows(LibraryFixture.Title("Karolcia"));
            fixture.SystemShows(LibraryFixture.Author("Maria Kruger"));
            fixture.SystemShows(LibraryFixture.Title("Renesans"));
            fixture.SystemShows(LibraryFixture.Author("Jerzy Konieczny"));
            fixture.SystemShowsAtLeastLines(fixture.StartBooksCount);
        }
    }
}

File: java/ArqLibraryClassic/ArqLibraryLayeredATests/src/bnsit/ms/arq/searching/SearchAll.java
package bnsit.ms.arq.searching;

import bnsit.ms.arq.library.LibraryFixture;
import org.junit.Before;
import org.junit.Test;

import static bnsit.ms.arq.library.LibraryFixture.author;
import static bnsit.ms.arq.library.LibraryFixture.title;

public class SearchAll {

    private LibraryFixture fixture = null;

    @Before
    public void setupFixture()
    {
        fixture = new LibraryFixture();
    }

    @Test
    public void shouldSearchAllBooks()
    {
        //given
        fixture.applicationStarted();
        fixture.hasSampleBooks();

        //when
        fixture.userEnters("search");

        //then
        fixture.then();
        fixture.systemShows(title("Karolcia"));
        fixture.systemShows(author("Maria Kruger"));
        fixture.systemShows(title("Renesans"));
        fixture.systemShows(author("Jerzy Konieczny"));
        fixture.systemShowsAtLeastLines(fixture.startBooksCount());
    }
}
